config/routes.rb

```diff
@@ -4,4 +4,9 @@
   resource :homepage, only: %i[show]
 
   root "homepages#show"
+
+  unless Rails.env.development? || Rails.env.test?
+    put "/rails/active_storage/disk/:encoded_token", to: redirect("/404")
+  end
+  post "/rails/active_storage/direct_uploads", to: redirect("/404")
```

Comment:
This should result in a routing error, not a redirect. That way the default rules apply.

Ideally we want to undefined the routes that ActiveStorage engine adds, but I don't know if that's possible. This is a hack approach that needs more thought imo.

Also needs documentation, as this is appropriate for Fringe and Koi but not generalisable to all situations (e.g. what if you have authenticated users and you want to allow them to upload files?

Finally, is this a realistic concern? Have the Rails team discussed why it's acceptable to allow arbitrary users to write to this endpoint? Is there any sort of nonce or CSFR protection that we're unaware of protecting these endpoints for DOS?

Reply:
There is an open rails issue about this:
https://github.com/rails/rails/issues/34961

A blog post with suggested workaround:
https://givenis.me/securing-rails-active-storage-direct-uploads

This blog post uses `authenticate_user`, but we could instead cause a 404 by raising a routing error?

Reply:
Another related open rails issue:
https://github.com/rails/rails/pull/52505

Reply:
- There is no way to remove a specific route; but you can remove all active storage routes by setting the `draw_routes` config option to `false` or you could overwrite the default active storage route in your routes file
- There is CSRF protection in DirectUploads but it is available on all the pages as a meta content